metricflow/model/dbt_mapping_rules/dbt_metric_to_dimensions_rules.py

```diff
@@ -132,6 +132,23 @@ def run(dbt_metrics: Tuple[MetricNode, ...], objects: MappedObjects) -> ModelVal
                                     extra_detail=f"columns: {','.join([column.name for column in metric.model.columns])}",
                                 )
                             )
+
+                        # try to build a dimension from filter.value
+                        # A `filter.value` can be a dimension, but doesn't have to be. If a
+                        # `filter.value` starts with a single quote, it's either a string or a
+                        # timestamp. Which then means a `filter.value` not beginning with a
+                        # single quote are either numbers, booleans, database functions, or a
+                        # column/dimension. Thus if a `filter.value` doesn't begin with a single
+                        # quote and matches a column name on the model, it's likely a dimension.
+                        # But it's not a match for a column on the model, it's likely a number,
+                        # boolean, function, etc and thus not a issue.
```

Comment:
The `COALESCE(valid_dimension, other_dimension)` case can still pop up here. Or even something like `normalized_value <> (raw_dimension_value * normalization_factor)`, where we only really know about `normalized_value` because the other inputs aren't listed as dimensions for any metric, but are included in the `columns` construct.

We could just add all columns as dimensions, but I don't know how reasonable that is.

Reply:
Resolved by: https://github.com/transform-data/metricflow/pull/324/commits/a18c1dc6e62f491c4da3339eb8052b802c3ae339